recipes/m4/all/conanfile.py

```diff
@@ -67,6 +68,19 @@ def generate(self):
                 "-rtlib=compiler-rt",
                 "-Wno-unused-command-line-argument",
             ])
+        if cross_building(self) and is_msvc(self):
+            triplet_arch_windows = {"x86_64": "x86_64", "x86": "i686", "armv8": "aarch64"}
+            
+            host_arch = triplet_arch_windows.get(str(self.settings.arch))
+            build_arch = triplet_arch_windows.get(str(self._settings_build.arch))
+
+            if host_arch and build_arch:
+                host = f"{host_arch}-w64-mingw32"
+                build = f"{build_arch}-w64-mingw32"
+                tc.configure_args.extend([
+                    f"--host={host}",
+                    f"--build={build}",
+                ])
```

Comment:
Yeah, I think this is the way at the moment.
It's true that this flow still leaves much to be desired 😞 
I'll open a ticket to improve this UX because, IMO, this could be something like:

```python
        if cross_building(self) and is_msvc(self):
            # tc.cross_build could contain all the information (like MesonToolchain)
            host_arch = tc.cross_build["host"]["arch"]
            build_arch = tc.cross_build["build"]["arch"]
            # Let's trust that AutotoolsToolchain is getting the correct GNU arch
            if host_arch and build_arch:
                tc.cross_build["host"]["triplet"] = f"{host_arch}-w64-mingw32"
                tc.cross_build["build"]["triplet"] = f"{build_arch}-w64-mingw32"
```
(very similar to MesonToolchain approach)